drivers/timer/npcx_itim_timer.c

```diff
@@ -95,18 +98,27 @@ static inline uint64_t npcx_itim_get_sys_cyc64(void)
 	}
 }
 
-static inline void npcx_itim_evt_enable(void)
+static inline int npcx_itim_evt_enable(void)
 {
+	uint64_t cyc_start;
+
 	/* Enable event timer and wait for it to take effect */
 	evt_tmr->ITCTS32 |= BIT(NPCX_ITCTSXX_ITEN);
 
 	/*
 	 * Usually, it need one clock (30.5 us) to take effect since
 	 * asynchronization between core and itim32's source clock (LFCLK).
 	 */
+	cyc_start = npcx_itim_get_sys_cyc64();
 	while (!IS_BIT_SET(evt_tmr->ITCTS32, NPCX_ITCTSXX_ITEN)) {
-		continue;
+		if (npcx_itim_get_sys_cyc64() - cyc_start >
+						NPCX_ITIM_EN_TIMEOUT_CYCLES) {
+			LOG_ERR("Timeout: enabling EVT timer!");
+			return -ETIMEDOUT;
```

Comment:
Return this error code up through this calling path: z_clock_driver_init()->npcx_itim_start_evt_tmr_by_tick()->npcx_itim_evt_enable().  The other paths don't look like they benefit from returning this error.

Reply:
Oops. I forgot the 'non-tickless' path. I will add them later.

Reply:
Done.